### PR TITLE
Move repl loop and command/script execution to nu_cli

### PR DESCRIPTION
**Pull request by chvck**
# Description

Within Couchbase Shell we need to be able to start the Nushell repl and execute scripts/files. By moving that functionality to nu_cli, this enables us to do so without duplicating a load of code. This moving could likely be more nuanced, there's a lot that has had to move here.

### CODE REVIEW

File: crates/nu-cli/src/config_files.rs [hunk 1]
```diff
@@ -92,7 +96,7 @@ pub(crate) fn read_config_file(
                 .read_line(&mut answer)
                 .expect("Failed to read user input");
 
-            let config_file = include_str!("../docs/sample_config/default_config.nu");
+            let config_file = include_str!("../../../docs/sample_config/default_config.nu");
```
Comment (chvck) [inline crates/nu-cli/src/config_files.rs:99]: I'm slightly dubious here, but it seems to be correct with regards to how it worked previously.

File: src/main.rs [hunk 2]
```diff
@@ -187,28 +180,34 @@ fn main() -> Result<()> {
             }
 
             if let Some(commands) = &binary_args.commands {
-                #[cfg(feature = "plugin")]
```
Comment (chvck) [inline src/main.rs:190]: Just noticed these removals, will add back.

**Comment by stormasm** [comment]
@chvck yes !  glad we have folks like you who are using nushell as a third party cli to your core product....  Other developers will want to do the same as well.  So its good we make this change now helping others down the road who want to talk to other systems besides couchbase...  Thanks for making the changes.   

Now we will have a much cleaner src/main.rs with more limited functionality ---- and put the repl logic where it should be down in the cli....  

Thank you !  good catch... and the way the "old" nushell (0.44) worked as well...

File: src/main.rs (hunk 2 above)
Comment (chvck) [reply src/main.rs:190]: Oh I remember now, moved these checks to the functions themselves.

File: crates/nu-cli/src/repl.rs [hunk 3]
```diff
@@ -20,9 +19,10 @@ use nu_protocol::{PipelineData, Spanned};
 use reedline::{DefaultHinter, Emacs, Vi};
 use std::{sync::atomic::Ordering, time::Instant};
 
-pub(crate) fn evaluate(
+pub fn evaluate_repl(
```
[the remark this answers is not in the document]
Comment (jntrnr) [reply crates/nu-cli/src/repl.rs:22]: I like the approach you're describing

File: crates/nu-cli/src/config_files.rs (hunk 1 above)
Comment (jntrnr) [reply crates/nu-cli/src/config_files.rs:99]: Yeah, include_str feels a bit magic, but I *think* so long as it works we're... okay?

Comment (chvck) [reply crates/nu-cli/src/config_files.rs:99]: It seems to work but I'm unsure how it works once nushell is distributed via something like homebrew? I don't fully understand how that works though... If that's fine then it solves a headache I have in couchbase shell too :)

Comment (chvck) [reply crates/nu-cli/src/config_files.rs]: Right, but is the path it refers to always valid?

Comment (jntrnr) [reply crates/nu-cli/src/config_files.rs]: yeah, since it just has to be be valid at build time, and it's relative to that subcrate.

**Comment by chvck** [comment]
I'm not entirely sure why CI is failing, the import is used.

**Comment by jntrnr** [comment]
This looks good - we can land and continue improving

**Pull request merged**